breaking: upgrade cy.readFile() to be a query command (#25595)
cypress-io/cypress: cli/CHANGELOG.md

<!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
## 13.0.0

_Released 01/31/2023 (PENDING)_

> **Breaking Changes**

Reviewer: ```suggestion
**Breaking Changes:**
```

> - `cy.readFile()` is now a [query command](https://on.cypress.io/retry-ability). This should not affect any tests using it; the functionality is unchanged. However, it can no longer be overwritten using `Cypress.Commands.overwrite()` This is a prerequisite to address [#25134](https://github.com/cypress-io/cypress/issues/25134). Addressed in [#25595](https://github.com/cypress-io/cypress/pull/25595).

Reviewer: ```suggestion
- The [`cy.readFile()`](/api/commands/readfile) command is now a [query command](https://on.cypress.io/retry-ability). This should not affect any tests using it; the functionality is unchanged. However, it can no longer be overwritten using [`Cypress.Commands.overwrite()`](/api/cypress-api/custom-commands#Overwrite-Existing-Commands). Addressed in [#25595](https://github.com/cypress-io/cypress/pull/25595).
```
This

> - The [`cy.readFile()`](/api/commands/readfile) command is now a [query command](https://on.cypress.io/retry-ability). This should not affect any tests using it; the functionality is unchanged. However, it can no longer be overwritten using [`Cypress.Commands.overwrite()`](/api/cypress-api/custom-commands#Overwrite-Existing-Commands). Addressed in [#25595](https://github.com/cypress-io/cypress/pull/25595).

Reviewer 2: Trying to highlight the benefit of the breaking change more. 🤷‍♂️ 

```suggestion
- The [`cy.readFile()`](/api/commands/readfile) command is now retry-able as a [query command](https://on.cypress.io/retry-ability). This should not affect any tests using it; the functionality is unchanged. However, it can no longer be overwritten using [`Cypress.Commands.overwrite()`](/api/cypress-api/custom-commands#Overwrite-Existing-Commands). Addressed in [#25595](https://github.com/cypress-io/cypress/pull/25595).
```

Author: :+1:
